docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md

```diff
@@ -144,15 +155,30 @@ Grafana supports two ways of specifying both the `certificate` and `private_key`
 You can only use one form of each configuration option. Using multiple forms, such as both `certificate` and `certificate_path`, results in an error.
 {{% /admonition %}}
 
----
+Always work with your company's security team on setting up certificates and private keys. If you need to generate them yourself (such as in the short term, for testing purposes, and so on), use the following example to generate your certificate and private key, including the step of ensuring that the key is generated with the [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.
 
-### Generate private key for SAML authentication:
+### Example of private key generation for SAML authentication
 
 An example of how to generate a self-signed certificate and private key that's valid for one year:
 
 ```sh
 $ openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes​
 ```
+Base64-encode the cert.pem and key.pem files:
+(-w0 switch is not needed on Mac, only for Linux)
+
+```sh
+$ base64 -w0 key.pem > key.pem.base64
+$ base64 -w0 cert.pem > cert.pem.base64
+```
+The base64-encoded values (`key.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.
+Run the following command to convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.
+
+```sh
+openssl pkcs8 -topk8 -inform PEM -outform PEM -in myoriginalkey.pem -out myconvertedkey.pem -nocrypt
+```
+
+The base64-encoded values (`myconvertedkey.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.
```

Comment:
It's not needed, because the private key will be in PKCS8 format.
```suggestion
Run the following command to convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

```sh
openssl pkcs8 -topk8 -inform PEM -outform PEM -in myoriginalkey.pem -out myconvertedkey.pem -nocrypt
```

The base64-encoded values (`myconvertedkey.pem.base64, cert.pem.base64` files) are then used for certificate and private_key.
```